email_totals/app.py

```diff
@@ -84,17 +159,97 @@ def get_account_totals(target_p, compare_p, minimum):
             222233334444:
                 total: 10
     ```
+
+    ```
+    111122223333: friendly-name
+    222233334444: account-two
+    ```
     """
 
+    def _build_result_dict(results_by_time):
+        """
+        Transform the results from cost explorer into a dictionary of the form:
+        ```
+        111122223333: 100.0
+        222233334444: 10
+        ```
+        """
+        account_totals = {}
+        for result in results_by_time:
+            for group in result['Groups']:
+                amount = float(group['Metrics']['UnblendedCost']['Amount'])
+
+                # Keys preserve the order defined in the GroupBy parameter from
+                # the call to get_cost_and_usage().
+                if len(group['Keys']) != 1:
+                    LOG.error(f"Unexpected grouping: {group['Keys']}")
+                    continue
+
+                # Add this account total to our output
+                account_id = group['Keys'][0]
+                if account_id not in account_totals:
+                    account_totals[account_id] = amount
+                else:
+                    LOG.error(f"Duplicate account total found: {account_id}")
```

Comment:
why would you ever get duplicates?

Reply:
that would mean cost explorer listed the account twice (I don't expect that to ever happen, but it might if the AWS service is degraded or something)